### PR TITLE
Hw 3

### PR DESCRIPTION
**Pull request by simyara**
Related to Homework 3

### CODE REVIEW

File: dist/utils/streams.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+"use strict";
```
Comment (abramenal) [inline dist/utils/streams.js]: Do we really need this file? Or we can completely add dist/ folder to gitignore?

**Review by abramenal — APPROVED** [review]

**Review by abramenal — DISMISSED** [review]

File: package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "NodeJS-Mentoring-HW3",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "nodemon dist/app.js",
+    "build": "babel src --out-dir dist",
+    "bStart": "npx babel-node src/app.js"
```
Comment (pavelauramenka) [inline package.json]: Any of the binaries loaded as dependencies and/or devDependencies are available in npm scripts directly (without any 3rd-parties like npx

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,7 @@
+node_modules
+npm-debug.log
+.idea
+package-lock.json
+dist/app.js
+dist/models/
+dist/modules/
```
Comment (pavelauramenka) [inline .gitignore]: Can't we ignore the whole `dist/` folder?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    "babel-preset-stage-2": "^6.24.1",
+    "babel-register": "^6.26.0",
+    "nodemon": "^1.17.3"
+  },
+  "bugs": {
+    "url": "https://github.com/simyara/NodeJS-Mentoring-HW3/issues"
+  },
+  "homepage": "https://github.com/simyara/NodeJS-Mentoring-HW3#readme",
+  "dependencies": {
+    "babel-promisify": "^1.0.0",
```
Comment (pavelauramenka) [inline package.json]: Isn't it a devDependency?

File: package.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  "keywords": [],
+  "author": "Yaraslauna Simaka",
+  "license": "ISC",
+  "devDependencies": {
+    "babel-cli": "^6.26.0",
+    "babel-core": "^6.26.3",
+    "babel-preset-env": "^1.7.0",
+    "babel-preset-stage-0": "^6.24.1",
+    "babel-preset-stage-2": "^6.24.1",
```
Comment (pavelauramenka) [inline package.json]: I think you don't use those state-* presets, so they can be removed

File: src/utils/streams.js [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+
+    }
+
+    convertToFile() {
+
+        if (!this.fileName) {
+            throw new Error(`File name not exist`);
+        }
+
+        import util from 'util';
```
Comment (pavelauramenka) [inline src/utils/streams.js]: Does it make sense to make the imports in the method body?

File: src/utils/streams.js [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+        import readline from 'readline';
+
+        let promises = [];
+
+        let downloadFile = function () {
+            return new Promise(function (resolve, reject) {
+                import https from 'https';
+
+                let file = fs.createWriteStream(`${path}nodejs-homework3.css`);
+                let request = https.get("https://doc-0g-0s-docs.googleusercontent.com/docs/securesc/ha0ro937gcuc7l7deffksulhg5h7mbp1/jsn8uecejvi98hj4v1vl2t1p830iaq44/1528106400000/05195043963524607236/*/1tCm9Xb4mok4Egy2WjGqdYYkrGia0eh7X?e=download", function(response) {
```
Comment (pavelauramenka) [inline src/utils/streams.js]: I think this constant value should be declared somewhere

File: package.json (hunk 2 above)
Comment (abramenal) [inline package.json]: Any of the binaries loaded as dependencies and/or devDependencies are available in npm scripts directly (without any 3rd-parties like npx)

File: .gitignore (hunk 3 above)
Comment (abramenal) [inline .gitignore]: Can't we ignore the whole `dist/` folder?

File: package.json (hunk 4 above)
Comment (abramenal) [inline package.json]: Isn't it a devDependency? Do you use it?

File: package.json (hunk 5 above)
Comment (abramenal) [inline package.json]: I think you don't use those stage-* presets, so they can be removed.

File: src/utils/streams.js [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+            console.log(err);
+        });
+    }
+
+    convertFromFile() {
+        if (!this.fileName) {
+            throw new Error(`File name not exist`);
+        }
+
+        import util from 'util';
```
Comment (abramenal) [inline src/utils/streams.js]: Does it make sense make the import s in the method's body?

File: src/utils/streams.js (hunk 7 above)
Comment (abramenal) [inline src/utils/streams.js]: I think this constant value should be declared somewhere.

**Review by abramenal — APPROVED** [review]

**Pull request merged**